packages/hardhat-core/test/builtin-tasks/flatten.ts

```diff
@@ -412,6 +415,18 @@ describe("Flatten task", () => {
       assert(!spyFunctionConsoleWarn.called);
     });
 
+    it("should write to an output file when the parameter output is specified", async function () {
+      const outputFile = "flatten.sol";
+      await this.env.run(TASK_FLATTEN, {
+        files: ["contracts/A.sol", "contracts/D.sol"],
+        output: outputFile,
+      });
+      const expected = await getExpectedSol();
+      const actual = readFileSync(outputFile, "utf8");
+      assert.equal(actual, expected);
+      removeSync(outputFile);
```

Comment:
Instead of creating the file in the cwd and then removing it, I think it's slightly better to create it in a temporary dir. You can do it like this:

```typescript
const outputFile = path.resolve(os.tmpdir(), "flattened.sol");
```

(Give or take; I haven't tried it)

Reply:
Good point :) 

Addressed with `const outputFile = `${tmpdir()}/flatten.sol`;`